src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -283,7 +244,7 @@ static void growSynapses(Connections &connections,
   // Check if we're going to surpass the maximum number of synapses. //TODO delegate this to createSynapse(segment)
   const size_t overrun = (connections.numSynapses(segment) + nActual - maxSynapsesPerSegment);
   if (overrun > 0) {
```

Comment:
please fix here what @dkeeney found in #194 : overrun can underflow in the minus operation. make it `int` here to fix.

Reply:
not sure if worth it, but would be nice to have a test to detect this type of error